src/decode.rs

```diff
@@ -28,6 +28,30 @@ impl From<io::Error> for TagDecodeError {
     }
 }
 
+impl Error for TagDecodeError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match self {
+            TagDecodeError::IOError { io_error } => Some(io_error),
+            _ => None,
+        }
+    }
+}
+
+impl Display for TagDecodeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use TagDecodeError::*;
```

Comment:
Why you put this import here?

Reply:
Just for the match, I can remove it if you want.